modules/core/src/lib/attribute.js

```diff
@@ -405,7 +417,10 @@ export default class Attribute extends BaseAttribute {
     for (const object of iterable) {
       objectInfo.index++;
 
-      const objectValue = accessorFunc(object, objectInfo);
+      let objectValue = accessorFunc(object, objectInfo);
+      if (transform) {
+        objectValue = transform.call(this, objectValue);
```

Comment:
Nit: Why do we call this as a method?  Why don't we just pass in `this` as the first param or require the app to bind it? 

This is not a pattern we use much...

Reply:
When we declare these attributes, the `transform` callbacks are assigned to methods of the layer. When the attributes are calculated (on data load), `this` in the callbacks still point to the old layer instance. This can be seen in the `IconLayer`.